core/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala

```diff
@@ -176,8 +186,61 @@ abstract class AppBase(
 
   var driverAccumMap: HashMap[Long, ArrayBuffer[DriverAccumCase]] =
     HashMap[Long, ArrayBuffer[DriverAccumCase]]()
-  var mlEventLogType = ""
-  var pysparkLogFlag = false
+
+
+  // Returns the String value of the eventlog or empty if it is not defined. Note that the eventlog
+  // won't be defined for running applications
+  def getEventLogPath: String = {
+    eventLogInfo.map(_.eventLog).getOrElse(new Path("")).toString
+  }
+
+  // Update the endTime of the application and calculate the duration.
+  // This is called while handling ApplicationEnd event
+  def updateEndTime(newEndTime: Long): Unit = {
+    appMetaData.foreach(_.setEndTime(newEndTime))
+  }
+
+  // Returns a boolean flag to indicate whether the endTime was estimated.
+  def isAppDurationEstimated: Boolean = {
+    appMetaData match {
+      case Some(meta) => meta.isDurationEstimated
+      case _ => true
+    }
+  }
+
+  // Returns the AppName
+  def getAppName: String = {
+    appMetaData.map(_.appName).getOrElse("")
+  }
+
+  // Returns optional endTime in ms.
+  def getAppEndTime: Option[Long] = {
+    appMetaData.flatMap(_.endTime)
+  }
+
+  // Returns optional wallClock duration of the Application
+  def getAppDuration: Option[Long] = {
+    appMetaData.flatMap(_.duration)
+  }
+
+  // Returns a boolean true/false. This is used to check whether processing an eventlog was
+  // successful.
+  def isAppMetaDefined: Boolean = appMetaData.isDefined
```

Comment:
the appMetaData honestly seems a bit weird to me. I know we had ApplicationCase before, but if you are accessing things from AppBase should those just exist in the AppBase, we have this weird split of things where some are in both but we seem to wrap but we can still access appMetaData directly.   I think its fine to leave for now but its confusing and we should better define the encapsulation